pkg/traces/servicegraphprocessor/processor.go

```diff
@@ -141,91 +153,96 @@ func newProcessor(nextConsumer consumer.Traces, cfg *Config) *processor {
-		Name:      "service_graph_request_total",
-		Help:      "Total count of requests between two nodes",
-	}, []string{"client", "server"})
-	p.serviceGraphRequestFailedTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "traces",
-		Name:      "service_graph_request_failed_total",
-		Help:      "Total count of failed requests between two nodes",
-	}, []string{"client", "server"})
-	p.serviceGraphRequestServerHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-		Namespace: "traces",
-		Name:      "service_graph_request_server_seconds",
-		Help:      "Time for a request between two nodes as seen from the server",
-		Buckets:   prometheus.ExponentialBuckets(0.01, 2, 12),
-	}, []string{"client", "server"})
-	p.serviceGraphRequestClientHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-		Namespace: "traces",
-		Name:      "service_graph_request_client_seconds",
-		Help:      "Time for a request between two nodes as seen from the client",
-		Buckets:   prometheus.ExponentialBuckets(0.01, 2, 12),
-	}, []string{"client", "server"})
-	p.serviceGraphUnpairedSpansTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "traces",
-		Name:      "service_graph_unpaired_spans_total",
-		Help:      "Total count of unpaired spans",
-	}, []string{"client", "server"})
-	p.serviceGraphDroppedSpansTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "traces",
-		Name:      "service_graph_dropped_spans_total",
-		Help:      "Total count of dropped spans",
-	}, []string{"client", "server"})
-
-	cs := []prometheus.Collector{
-		p.serviceGraphRequestTotal,
-		p.serviceGraphRequestFailedTotal,
-		p.serviceGraphRequestServerHistogram,
-		p.serviceGraphRequestClientHistogram,
-		p.serviceGraphUnpairedSpansTotal,
-		p.serviceGraphDroppedSpansTotal,
+func (p *processor) registerMetrics(mp metric.MeterProvider, meterId string) error {
+	meter := mp.Meter(meterId)
+
+	var err error
+	p.serviceGraphRequestTotal, err = meter.Float64Counter(
+		serviceGraphRequestTotal_name,
+		metric.WithDescription("Total count of requests between two nodes"),
+	)
+	if err != nil {
+		return err
 	}
 
-	for _, c := range cs {
-		if err := p.reg.Register(c); err != nil {
-			return err
-		}
+	p.serviceGraphRequestFailedTotal, err = meter.Float64Counter(
+		serviceGraphRequestFailedTotal_name,
+		metric.WithDescription("Total count of failed requests between two nodes"),
+	)
+	if err != nil {
+		return err
 	}
 
-	return nil
-}
+	p.serviceGraphRequestServerHistogram, err = meter.Float64Histogram(
+		serviceGraphRequestServerHistogram_name,
+		metric.WithDescription("Time for a request between two nodes as seen from the server"),
+	)
+	if err != nil {
+		return err
+	}
+
+	p.serviceGraphRequestClientHistogram, err = meter.Float64Histogram(
+		serviceGraphRequestClientHistogram_name,
+		metric.WithDescription("Time for a request between two nodes as seen from the client"),
+	)
+	if err != nil {
+		return err
+	}
```

Comment:
We use our own fork, so theoretically anything is possible :) From a technical point of view we could migrate while still using the Prometheus instrumentation.

The reason I opted for Otel is that I wanted to avoid using a fork of Collector as much as possible. A PR like [this](https://github.com/open-telemetry/opentelemetry-collector/pull/7696) would be almost certainly rejected if it exposes ways of configuring telemetry which is not via the Otel SDK. Even if the PR doesn't get merged, it is likely that we will reduce the maintenance cost of our fork if we stay more in line with the Otel way of doing things.

Is there something specific about moving to Otel instrumentation which bothers you? Personally I thought it should be enough to just run an old an a new version of the Agents and see if any metrics are behaving differently. In my local testing I tried to enable as many things as possible so I get as many metrics as possible. Personally the things that bother me are:

* How to make sure I enabled all the components and features which could possible generate metrics?
* There are "dynamic" metrics like the service graph ones which are not visible when the Agent starts. They are only seen later, when we hit the exact part of the code which generates the metric. I'll have to make sure all the code paths that generate metrics are hit somehow.

Reply:
> Is there something specific about moving to Otel instrumentation which bothers you?

Mainly it's because the entire rest of the code base uses Prometheus for metrics. Not only do I think the Prometheus API is simpler for metrics, but it's also well-understood by us, so having some code using the OpenTelemetry SDK for metrics would cause some inconsistency and small knowledge gaps. 

I don't want to block the PR on this though, especially since this is static mode, so I'll leave the choice up to you 👍